src/seq_analysis/orf.rs

```diff
@@ -137,28 +143,34 @@ where
             self.state.codon.push_back(*nuc.borrow());
             offset = (index + 1) % 3;
 
+            // check if entering orf
+            if self.finder.start_codons.contains(&self.state.codon) {
+                self.state.start_pos[offset].push(index);
+            }
```

Comment:
No longer can be an `else if`, since you can be in an ORF and find a new starting site now. I've moved this to the top here (before checking if we are currently in an ORF) just for clarity when reading